cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -611,11 +611,13 @@ updateOutputTemplate
     -- If the initial funds are equal to the maximum funds, rewards cannot be created.
     subtractForTreasury :: Integer
     subtractForTreasury = nonDelegCoin `quot` 10
-    totalSupply, nonDelegCoin, delegCoin :: Integer
+    totalSupply :: Word64
     -- if --total-supply is not specified, supply comes from the template passed to this function:
-    totalSupply = fromIntegral (maybe maximumLovelaceSupply unLovelace mTotalSupply)
-    delegCoin = case mDelegatedSupply of Nothing -> 0; Just amountDeleg -> totalSupply - unLovelace amountDeleg
-    nonDelegCoin = totalSupply - delegCoin
+    totalSupply = maybe maximumLovelaceSupply unLovelace mTotalSupply
+    totalSupply', delegCoin, nonDelegCoin :: Integer
+    totalSupply' = fromIntegral totalSupply
```

Comment:
I would remove `totalSupply'` and call `fromIntegral` within the definitions of other variables.

Reply:
Done :+1: